components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -2356,9 +2347,72 @@ void RewardsServiceImpl::OnRestorePublishersInternal(
     ledger::OnRestoreCallback callback,
     bool result) {
   callback(result);
+}
+
+std::unique_ptr<ledger::PublisherInfoList>
+SaveNormalizedPublisherListOnFileTaskRunner(PublisherInfoDatabase* backend,
+                                            const ledger::PublisherInfoList& list) {
+  if (!backend) {
+    return nullptr;
+  }
+
+  bool success = backend->InsertOrUpdateActivityInfos(list);
 
-  if (result) {
-    TriggerOnContentSiteUpdated();
+  if (!success) {
+    return nullptr;
+  }
+
+  std::unique_ptr<ledger::PublisherInfoList> new_list(
+      new ledger::PublisherInfoList);
+
+  for (auto& publisher : list) {
+    new_list->push_back(publisher);
+  }
+
+  return new_list;
+}
+
+void RewardsServiceImpl::SaveNormalizedPublisherList(
+      const ledger::PublisherInfoListStruct& list) {
+
+  if (list.list_.size() == 0) {
+    std::unique_ptr<ledger::PublisherInfoList> empty_list(
+        new ledger::PublisherInfoList);
+    OnNormalizedPublisherListSaved(std::move(empty_list));
+    return;
+  }
+
+  base::PostTaskAndReplyWithResult(
+    file_task_runner_.get(),
+    FROM_HERE,
+    base::Bind(&SaveNormalizedPublisherListOnFileTaskRunner,
+               publisher_info_backend_.get(),
+               list.list_),
+    base::Bind(&RewardsServiceImpl::OnNormalizedPublisherListSaved,
+               AsWeakPtr()));
+}
+
+bool sortPublisherByPercentage(ContentSite a, ContentSite b) {
+  return a.percentage > b.percentage;
+}
+
+void RewardsServiceImpl::OnNormalizedPublisherListSaved(
+    std::unique_ptr<ledger::PublisherInfoList> list) {
+  if (!list) {
+    LOG(ERROR) << "Problem saving normalized publishers "
```

Comment:
we can't just log this error and ignore it here, `SaveNormalizedPublisherList` needs a callback

Reply:
sure we can add callback to go back to native ledger. What should we do in native ledger with this callback?